### PR TITLE
media-libs/kvazaar: EAPI=7, test deps, fix tests

### PR DESCRIPTION
**Pull request by candrews**
See upstream issue at https://github.com/ultravideo/kvazaar/issues/216

Bug: https://bugs.gentoo.org/671146
Package-Manager: Portage-2.3.52, Repoman-2.3.12
Signed-off-by: Craig Andrews <candrews@gentoo.org>

### CODE REVIEW

**Comment by a17r** [comment]
why not mask USE=test on these arches instead

**Comment by candrews** [comment]
> why not mask USE=test on these arches instead

There are still tests that run without valgrind, so IMHO it's better to not mask USE=test on any arch instead removing the tests requiring valgrind (which is what I did here in this PR). Even on a valgrind supporting arch, the valgrind tests fail because Gentoo doesn't have a sufficiently old version of ffmpeg which allows the tests to pass (which is what this issue is reporting).

File: media-libs/kvazaar/kvazaar-9999.ebuild [hunk 1]
```diff
@@ -24,9 +24,13 @@ SLOT="0/4"
 IUSE="static-libs test"
 REQUIRED_USE="test? ( static-libs )"
 
+RESTRICT="!test? ( test )"
```
Comment (aballier) [inline media-libs/kvazaar/kvazaar-9999.ebuild]: what's this ? I seem to recall some logic behind that but RESTRICT="test" seems simpler

**Comment by candrews** [comment]
`RESTRICT="!test? ( test )"` means running the tests requires that the test use flag be set. That way, we can be sure that if tests are run, that the test dependencies are present. There are many ebuilds in the tree with this expressions.

`RESTRICT="test"` would mean do not run tests. Which is not what we want to do here.

**Comment by aballier** [comment]
> `RESTRICT="!test? ( test )"` means running the tests requires that the test use flag be set. That way, we can be sure that if tests are run, that the test dependencies are present. There are many ebuilds in the tree with this expressions.


Ah, that's it, now I remember.
This has always been implicitly handled though: We do not want to skip tests because the useflag is missing, we want the useflag to be forcibly enabled when tests are to be run.

**Pull request closed without merging**